tests/bpfbytecode.cpp

```diff
@@ -20,11 +18,11 @@ BpfBytecode codegen(const std::string &input)
                 .put<BPFtrace>(*bpftrace)
                 .add(ast::AllParsePasses())
                 .add(ast::CreateSemanticPass())
+                .add(ast::AllCompilePasses())
                 .run();
   EXPECT_TRUE(ok && ast.diagnostics().ok());
-
-  ast::CodegenLLVM codegen(ast, *bpftrace);
-  return codegen.compile();
+  auto &output = ok->get<BpfBytecode>();
+  return std::move(output);
```

Comment:
```suggestion
  return ok->get<BpfBytecode>();
```
Why does this not work? The return type doesn't include a reference.

Reply:
I actually want to move out the value. (Essentially we leave an empty `BpfBytecode` inside the context to be destroyed.)

I could have a `std::move(ok->get<BpfBytecode>())` instead, but for some reason I thought that was less clear? Should I change this?

You definitely don't want to return a reference here, since that will be deleted once the context goes out of scope.

Reply:
Oh, I think I see your question now. I think what you're saying *would* be fine, if `BpfBytecode` had a suitable copy constructor. Unfortunately that is deleted (https://github.com/bpftrace/bpftrace/blob/master/src/bpfbytecode.h#L29), presumably to avoid accidental expensive copies.

Reply:
something something copy elision if you just do `ok->get<BpfBytecode>();` ? I need to read into this more.

Reply:
That only applies to temporary local values. You definitely won't be able to avoid copy/move elision in this case, since you're moving it out of the context.